src/index.ts

```diff
@@ -0,0 +1,101 @@
+import { TASK_FLATTEN_GET_FLATTENED_SOURCE } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { extendEnvironment, task } from "@nomiclabs/buidler/config";
+import { BuidlerPluginError, lazyObject } from "@nomiclabs/buidler/plugins";
+
+import AbiEncoder from "./AbiEncoder";
+import ContractCompiler from "./ContractCompiler";
+import EtherscanService from "./etherscan/EtherscanService";
+import EtherscanVerifyContractRequest from "./etherscan/EtherscanVerifyContractRequest";
+import SolcVersions from "./solc/SolcVersions";
+
+export class EtherscanBuidlerEnvironment {
+  constructor(
+    public readonly url: string = "https://api.etherscan.io/api",
+    public readonly token: string = ""
+  ) {}
+}
+
+declare module "@nomiclabs/buidler/types" {
+  export interface BuidlerRuntimeEnvironment {
+    etherscan: EtherscanBuidlerEnvironment;
+  }
+
+  export interface ResolvedBuidlerConfig {
+    etherscan: {
+      url?: string;
```

Comment:
Can this really be undefined? If no url is given the EtherscanBuidlerEnvironment will default to ```https://api.etherscan.io/api ``` on its constructor.